### DISCUSSION TITLE
Is there a way to refresh static files without server restart?

### DISCUSSION THREAD

**Discussion by pioz** [vercel/next.js#14384] [Help]
I've put online a new [next.js][1] app bootstrapped with *Create Next App*. I've made a build with `next build` command and start the app with `next start`.

Now, if I add a new file in my `public/` directory I get 404 until I restart my next.js app. Is there a way to refresh static files without server restart?

  [1]: https://nextjs.org/

**Comment by IamLizu** [comment]
When you build your application it creates a static optimized version of your application, changes made after a build needs to be built again to take affect on next build. Which means you gotta build every time.

If you are developing your site, use `npm run dev` instead to get hot-reload.

**Comment by AntonAndreevichMoroz** [comment]
Greetings. I have the same question. The situation is that my application displays a random image from the public/img folder on the page. Files in public/img are periodically added and removed. In my script, fs.readdir sees changes to the file system, but Nextjs does not update the public content until the restart. Accordingly, new added pictures in public/img display 404. Pictures can be added to this folder by administrators or automatically from another system. Is there a workaround to not reload Nextjs in production?
